docs/docs/reference-react-component.md

```diff
@@ -0,0 +1,170 @@
+
+```js
+constructor(props) {
+  super(props);
+  this.state = {
+    color: props.initialColor
+  };
+}  
+```
+
+### `.componentDidMount()`
+
+TODO
+
+### `.componentDidUpdate(prevProps, prevState)`
+
+TODO
+
+### `.componentWillMount()`
+
+TODO
+
+### `.componentWillReceiveProps(nextProps)`
+
+TODO
+
+### `.componentWillUnmount()`
+
+TODO
+
+### `.componentWillUpdate(nextProps, nextState)`
+
+TODO
+
+### `.defaultProps`
+
+`defaultProps` can be defined as a property on the component class itself, to set the default props for the class. For example:
+
+```js
+class CustomButton extends React.Component {
+  // ...
+}
+
+CustomButton.defaultProps = {
+  color: 'blue'
+};
+```
+
+If `props.color` is not provided, it will be set by default to `'blue'`.
+
+### `.displayName`
+
+The `displayName` string is used in debugging messages. JSX sets this value automatically; see [JSX in Depth](/react/docs/jsx-in-depth.html).
+
+### `.forceUpdate(callback)`
+
+TODO
+
+### `.props`
+
+`this.props` contains the props that were defined by the caller of this component. See [Components and Props](/react/docs/components-and-props.html) for an introduction to props.
+
+In particular, `this.props.children` is a special prop, typically defined by the child tags in the JSX expression rather than in the tag itself.
+
+### `.propTypes`
+
+`propTypes` can be defined as a property on the component class itself, to define what types the props should be. It should be a map from prop names to types as defined in `React.PropTypes`. In development mode, when an invalid value is provided for a prop, a warning will be shown in the JavaScript console. In production mode, `propTypes` checks are skipped for efficiency.
+
+For example, this code ensures that the `color` prop is a string:
+
+```js
+class CustomButton extends React.Component {
+  // ...
+}
+
+CustomButton.propTypes = {
+  name: React.PropTypes.string
+};
+```
+
+### `.render()`
+
+The `render()` method is required.
+
+When called, it should examine `this.props` and `this.state` and return a single React element. This element can be either a representation of a native DOM component, such as `<div />`, or another composite component that you've defined yourself.
+
+You can also return `null` or `false` to indicate that you don't want anything rendered. React will just render a comment tag. When returning `null` or `false`, `ReactDOM.findDOMNode(this)` will return `null`.
+
+The `render()` function should be pure, meaning that it does not modify component state, it returns the same result each time it's invoked, and it does not directly interact with the browser. If you need to interact with the browser, perform your work in `componentDidMount()` or the other lifecycle methods instead. Keeping `render()` pure makes components easier to think about.
+
+### `.setState(nextState, callback)`
+
+```javascript
+void setState(
+  function|object nextState,
```

Comment:
I thought we were going to avoid using this Flow/type-like notation in the docs.